airflow/executors/local_kubernetes_executor.py

```diff
@@ -0,0 +1,204 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Dict, List, Optional, Set, Union
+
+from airflow.configuration import conf
+from airflow.executors.base_executor import CommandType, EventBufferValueType, QueuedTaskInstanceType
+from airflow.executors.kubernetes_executor import KubernetesExecutor
+from airflow.executors.local_executor import LocalExecutor
+from airflow.models.taskinstance import SimpleTaskInstance, TaskInstance, TaskInstanceKey
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class LocalKubernetesExecutor(LoggingMixin):
+    """
+    LocalKubernetesExecutor consists of LocalExecutor and KubernetesExecutor.
+    It chooses an executor to use based on the queue defined on the task.
+    When the queue is the value of ``kubernetes_queue`` in section ``[local_kubernetes_executor]``
+    of the configuration (default value: `kubernetes`), KubernetesExecutor is selected to run the task,
+    otherwise, LocalExecutor is used.
+    """
+
+    supports_ad_hoc_ti_run: bool = True
```

Comment:
i think you need also to update the `flash` message [here](https://github.com/apache/airflow/blob/b5c0158b2eb646eb1db5d2c094d3da8f88a08a8b/airflow/www/views.py#L1607-L1609)

Reply:
on a related note... according to [that message](https://github.com/apache/airflow/blob/b5c0158b2eb646eb1db5d2c094d3da8f88a08a8b/airflow/www/views.py#L1607-L1609), celery / CKE, and KE all support "ad hoc" run.  why doesn't LocalExecutor? it looks like it's because it starts up an instance of the executor from within the webserver.  i'm surprised this works with k8s because i thought i recalled some issues where the webserver didn't have the right roles to create pods...  but that's another story.   Anyway, perhaps we need to add some way, more sophisticated now than `supports_ad_hoc_ti_run`, such that the ad hoc trigger  will work for tasks with k8s queue but not for local.  thoughts @kaxil ?

Reply:
Will try the  supports_ad_hoc_ti_run setting

Reply:
Yeah @dstandish I think running ad-hocs TIs from Webserver is semi-broken -- I would vote for removing or using "clearing" mechanism under the hood so a new instance of Executor does not need to be created.

Reply:
> Yeah @dstandish I think running ad-hocs TIs from Webserver is semi-broken -- I would vote for removing or using "clearing" mechanism under the hood so a new instance of Executor does not need to be created.

OK so should @subkanthi just leave this as is for now and we deal with the  webserver bit later?

Reply:
Yeah

Reply:
@dstandish i guess I don't have to update the flash message, correct?

Reply:
yup